instance/tests/models/test_openedx_monitoring_mixins.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2016 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+OpenEdXInstance monitoring mixins - tests
+"""
+
+# Imports #####################################################################
+
+from unittest.mock import call, patch
+from uuid import uuid4
+
+from django.test import override_settings
+
+from instance.tests.base import TestCase
+from instance.tests.models.factories.openedx_instance import OpenEdXInstanceFactory
+from instance.tests.utils import patch_services
+
+
+# Tests #######################################################################
+
+@override_settings(ADMINS=[('OpenCraft Admin', 'admin@opencraft.com')],
+                   NEWRELIC_ADMIN_USER_API_KEY='admin-api-key')
+class OpenEdXMonitoringTestCase(TestCase):
+    """
+    Tests for OpenEdXMonitoringMixin.
+    """
+    @patch_services
+    @patch('instance.models.mixins.openedx_monitoring.OpenEdXMonitoringMixin.enable_monitoring')
+    def test_set_appserver_active(self, mocks, mock_enable_monitoring):
+        """
+        Check that monitoring is enabled when an appserver is activated.
+        """
+        instance = OpenEdXInstanceFactory()
+        appserver_id = instance.spawn_appserver()  # pylint: disable=no-member
```

Comment:
@omarkhan I see that in other tests we set `pylint: disable=no-member` at the top of the test class, but that doesn't seem like a great idea. I didn't try this, but maybe adding `OpenEdXInstanceFactory` to [ignored-classes in pylinrc](https://github.com/open-craft/opencraft/blob/05578e947b97fcc5ce0bbd44ef53c848524e0c30/pylintrc#L290) would work to silence these annoying `no-member` warnings without having to manually do it on every line?